### PR TITLE
Svtdanny/kvcache attention

### PR DESCRIPTION
**Pull request by svtdanny**

### CODE REVIEW

**Review by Muxas — COMMENTED** [review]
Rework some `clear_async`'s

File: wrappers/python/nntile/layer/attention.py [hunk 1]
```diff
@@ -142,6 +143,16 @@ def __init__(self, x_q: TensorMoments, x_k: TensorMoments, \
         else:
             self.redux = 0
 
+        self.reset_cache()
+
+        # need to fill with valid values for dynamic api usage
```
[the remark this answers is not in the document]
Comment (svtdanny) [reply wrappers/python/nntile/layer/attention.py:148]: For some reason it breaks the code. For me, it is also some strange

File: wrappers/python/nntile/layer/attention.py [hunk 2]
```diff
@@ -437,6 +484,32 @@ def forward_async(self):
...
+            add_fiber_async(1, self.in_proj_bias_k.value, 1, k_partial, 0, 1)
+
+        copy_intersection_async(k_partial, [0,self.k_cache_size,0,0], self.k.value, [0,0,0,0])
+        self.k_cache_size+=x.shape[1]
+
+        if use_kv_partials:
+            cached_shape = self.k.value.shape
+            cached_shape[1] = self.k_cache_size
+            k_partial_cached = nntc.zeros(cached_shape, dtype=type(x))
+            copy_intersection_async(self.k.value, [0,0,0,0], k_partial_cached, [0,0,0,0])
```
[the remark this answers is not in the document]
Comment (svtdanny) [reply wrappers/python/nntile/layer/attention.py]: yes. But for now because of softmax - no better choice

**Review by Muxas — APPROVED** [review]
This PR reorganises forward and backward operations into much more readable chunks and adds a prototype for a dynamically allocated variant

**Pull request merged**